src/ashley/models.py

```diff
@@ -64,3 +70,100 @@ class Meta:
 
 # Default implementation of the AbstractUser, using django-machina's model factory
 User = model_factory(AbstractUser)
+
+
+class AbstractLTIContext(Model):
+    """Abstract LTI Context model for Ashley."""
+
+    GROUP_PREFIX = "cg"
+    GROUP_DELIMITER = "__"
+
+    remote_id = models.CharField(max_length=150, null=False, blank=False, db_index=True)
+
+    lti_consumer = models.ForeignKey(
+        LTIConsumer, on_delete=models.CASCADE, null=False, db_index=True
```

Comment:
I find it dangerous and probably useless to cascade deletion:
```suggestion
        LTIConsumer, on_delete=models.RESTRICT, null=False, db_index=True
```

Reply:
I agree with you, I changed it to `on_delete=models.PROTECT`